docroot/themes/humsci/su_humsci_admin/js/node-protection.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @file
+ * Stops page from changing when user is posting.
+ */
+
+(function ($, Drupal) {
+  'use strict';
+  // Allow Save Submit button.
+  var save = false;
+  // Flag the form has been edited.
+  var edit = false;
+
+  Drupal.behaviors.nodeEditProtection = {
+    attach: function (context, settings) {
+      // Add listeners for all types of data entry.
+      $("input, button, :input", context).on('blur change click dblclick keydown mousedown select submit', function (e) {
+        edit = true;
+        save = false;
+
+        // Node save button.
+        if ($(this).attr('value') == 'Save') {
+          save = true;
+        }
+      });
+
+      // Handle backbutton, exit etc.
+      window.onbeforeunload = function () {
+        // Add CKEditor support.
+        if (typeof (CKEDITOR) != 'undefined' && typeof (CKEDITOR.instances) != 'undefined') {
```

Comment:
So, only if there is a wysiwyg on a content type will this functionality execute? What about pages with markdown or large amount of text fields?

Reply:
This sets the edit to true if the user changes anything in the ckeditor. since a user doesn't directly modify an <input> field or anything this helps with that support.
all other text fields and buttons are supported with the .on('blur', 'click' listener